src/libraries/Common/src/SourceGenerators/CSharpSyntaxUtilities.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Globalization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace SourceGenerators;
+
+internal static class CSharpSyntaxUtilities
+{
+    // Standard format for double and single on non-inbox frameworks.
+    public const string DoubleFormatString = "G17";
+    public const string SingleFormatString = "G9";
+
+    // Format a literal in C# format -- works around https://github.com/dotnet/roslyn/issues/58705
+    public static string FormatLiteral(object? value, TypeRef type)
+    {
+        if (value == null)
+        {
+            return $"default({type.FullyQualifiedName})";
+        }
+
+        if (type.TypeKind is TypeKind.Enum)
+        {
+            // Return the numeric value.
+            return FormatNumber();
+        }
+
+        switch (value)
+        {
+            case string @string:
+                return SymbolDisplay.FormatLiteral(@string, quote: true); ;
+            case char @char:
+                return SymbolDisplay.FormatLiteral(@char, quote: true);
+            case double.NegativeInfinity:
+                return "double.NegativeInfinity";
+            case double.PositiveInfinity:
+                return "double.PositiveInfinity";
+            case double.NaN:
+                return "double.NaN";
+            case double @double:
+                return $"{@double.ToString(DoubleFormatString, CultureInfo.InvariantCulture)}D";
+            case float.NegativeInfinity:
+                return "float.NegativeInfinity";
+            case float.PositiveInfinity:
+                return "float.PositiveInfinity";
+            case float.NaN:
+                return "float.NaN";
+            case float @float:
+                return $"{@float.ToString(SingleFormatString, CultureInfo.InvariantCulture)}F";
+            case decimal.MaxValue:
+                return "decimal.MaxValue";
```

Comment:
If `Min`/`MaxValue` are here, why not the same for cases like `double` or `float`?

Reply:
This was already existing in Json -- my understanding is that it was part of the workaround for https://github.com/dotnet/roslyn/issues/58705

Reply:
I think we can remove this special case, mentioned https://github.com/dotnet/runtime/pull/93160/files/c99abe6de78ea5cb63ddad48de7014ec7f6768a0#r1350960069